### DISCUSSION TITLE
async function in props of getServerSideProps

### DISCUSSION THREAD

**Discussion by Nithanaroy** [vercel/next.js#32153] [Help]
### What version of Next.js are you using?

12

### What version of Node.js are you using?

14

### What browser are you using?

Chrome

### What operating system are you using?

mac

### How are you deploying your application?

localhost

### Describe the Bug

```js
export async function getServerSideProps() {
    return { props: async () => {
        const res = await fetch(`...`)
        const data = await res.json()
        return data
    } }
}
```
returns
```
Error: Error serializing props returned from `getServerSideProps` in "/page1".
Reason: Props must be returned as a plain object from getServerSideProps: `{ props: { ... } }` (received: `[object AsyncFunction]`).
```

### Expected Behavior

Flush the head and wait for the data

### To Reproduce

Create a component with the above getServerSideProps function that uses async function as a return value for props.

**Comment by balazsorban44** [comment]
The error message says it all. You must return JSON serializable data. A function isn't that. run that logic before the return and only return the `data` instead

**Comment by Nithanaroy** [comment]
How can I achieve streaming response in next.js? One approach is shown in https://youtu.be/Nl4OwNhh2QI?t=1241

<img width="976" alt="Screen Shot 2021-12-06 at 6 55 50 AM" src="https://user-images.githubusercontent.com/670556/144773111-33e4682b-4988-441f-a012-a3f8a1114c05.png">

**Comment by Nithanaroy** [comment]
If the above is unavailable does next.js send the response headers immediately and not wait until the below promise is resolved (for the given below case) in SSR? How can I verify this?

```js
export async function getServerSideProps() {
    const res = await fetch(`...`)
    const data = await res.json()
    return {
        props: data
    }
}
```
